components/button/LinkButton.js

```diff
@@ -0,0 +1,85 @@
+import React, { PureComponent } from 'react';
+import PropTypes from 'prop-types';
+import cx from 'classnames';
+import theme from './theme.css';
+
+class LinkButton extends PureComponent {
+  static propTypes = {
+    children: PropTypes.any,
+    className: PropTypes.string,
+    disabled: PropTypes.bool,
+    element: PropTypes.element,
+    icon: PropTypes.element,
+    iconPlacement: PropTypes.oneOf(['left', 'right']),
+    inverse: PropTypes.bool,
+    label: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
+    onMouseLeave: PropTypes.func,
+    onMouseUp: PropTypes.func,
+    size: PropTypes.oneOf(['small', 'medium', 'large']),
+  };
+
+  static defaultProps = {
+    className: '',
+    iconPlacement: 'left',
+    inverse: false,
+    size: 'medium',
+  };
+
+  handleMouseUp = event => {
+    this.buttonNode.blur();
+    if (this.props.onMouseUp) {
+      this.props.onMouseUp(event);
+    }
+  };
+
+  handleMouseLeave = event => {
+    this.buttonNode.blur();
+    if (this.props.onMouseLeave) {
+      this.props.onMouseLeave(event);
+    }
+  };
+
+  render() {
+    const { children, className, disabled, element, icon, iconPlacement, inverse, label, size, ...others } = this.props;
+
+    const Element = element || 'button';
+
+    const classNames = cx(
+      theme['link-button'],
+      {
+        [theme['has-icon-only']]: !label && !children,
+        [theme['is-disabled']]: disabled,
+        [theme['is-inverse']]: inverse,
+        [theme[size]]: theme[size],
+      },
+      className,
+    );
+
+    const props = {
+      ...others,
+      ref: node => {
+        this.buttonNode = node;
+      },
+      className: classNames,
+      onMouseUp: this.handleMouseUp,
+      onMouseLeave: this.handleMouseLeave,
+      'data-teamleader-ui': 'link-button',
+    };
+
+    return React.createElement(
+      Element,
+      props,
+      icon && iconPlacement === 'left' && icon,
+      label || children ? (
```

Comment:
Q: Could you translate these lines to English so I can just understand what's happening here? I guess I don't know enough of these shorthands and can't really figure out what's happening.

Reply:
It's just a ternary operator, but it's just a bit confusing because of the multine and the OR in the condition.

it's actually just the same as this
```jsx
if (label || children) {
  return <span className={theme['children']}>{label}{children}</span>
}
return null;
```

 :ok: by @lowiebenoot on 04/01/2018 17:02:57

Reply:
Pretty funny you just answered this, I just understood it by looking at it again. My confusion was the comma on line 72, I thought `label || children ?` was connected to the `&&` on line 72, but it's not. Was just about to comment I finally understood :) Thx though

 :ok: by @InstaK on 04/01/2018 17:06:57

Reply:
```js